<?xml version="1.0"?>
<launch>

	<arg name="namespace" default="" />
	<arg name="publishFreq" default="30"/>
	<arg name="camTopic" default="camera/image_rect" />
	<arg name="calibFile" default="$(find tum_ardrone)/camcalib/mambo.txt" />
	<arg name="autorepeat_rate" default="30" />
	<arg name="deadzone" default="0.1" />
	<arg name="coalesce_interval" default=".001" />
		
	<group ns="$(arg namespace)/">
	
		<node name="drone_stateestimation" pkg="tum_ardrone" type="drone_stateestimation" output="screen">	  
			<param name="calibFile" type="string" value="$(arg calibFile)" />
			<param name="publishFreq" type="double" value="$(arg publishFreq)" />	
			<param name="camTopic" type="string" value="$(arg camTopic)" />  
			<param name="ns" type="string" value="$(arg namespace)" />
		</node>	

		<node name="drone_autopilot" pkg="tum_ardrone" type="drone_autopilot" output="screen">
		</node>

		<node name="drone_gui" pkg="tum_ardrone" type="drone_gui" output="screen">
		</node>

		<node name="dynamic_param" pkg="rqt_reconfigure" type="rqt_reconfigure">
		</node>
  
		<node name="joy_node" pkg="joy" type="joy_node">
			<param name="autorepeat_rate" type="double" value="$(arg autorepeat_rate)" />
			<param name="deadzone" type="double" value="$(arg deadzone)" />
			<param name="coalesce_interval" type="double" value="$(arg coalesce_interval)" />
		</node> 

	</group>

<!--	<node name="world_map" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 world $(arg namespace)/map 100" />
-->

 </launch>
